### ISSUE TITLE
Tiled Galleries: Blurry images on the carousel

### ISSUE THREAD

**Issue by chaitanyamsv** [Automattic/jetpack#6625]
Images on the gallery carousel are being stretched, losing clarity and becoming blurry.

#### Steps to reproduce the issue
Click on a gallery image, so it opens on the carousel.

#### What I expected
Clear images with the same clarity as they were on the gallery.

#### What happened instead
Images on the gallery carousel are being stretched, losing clarity and becoming blurry.

Example sites: 
https://saramakdessi.com/illustrations/
https://hfxart.com/

Chats and tickets: 215297-hc 3100136-t 233958-chat  233966-chat 

Possibly related to p1489076574206964-slack-triage

**Comment by georgestephanis** [comment]
I think I've fixed it wpcom-side with r152553-wpcom

I believe the cause of this traces back to https://github.com/Automattic/jetpack/pull/3680

**Comment by withinboredom** [comment]
@georgestephanis 

This looks like an issue with image sizes, actually:

Current img tag example with irrelevant parts stripped and original url removed:

```html
<img
data-orig-file="img.jpg" 
data-orig-size="2048,1365"
data-medium-file="img.jpg?w=300" 
data-large-file="img.jpg?w=1000"
src="https://i1.wp.com/img.jpg?w=524&h=349&crop&ssl=1"
width="524"
height="349" 
data-original-width="524"
data-original-height="349"
itemprop="http://schema.org/image"
style="width: 162px; height: 107px;"
srcset="https://i1.wp.com/img.jpg?w=524&h=349&crop&ssl=1&zoom=2 2x"
scale="2">
```

All the attributes look correct except `data-medium-file` and `data-large-file`, which are lacking the height and crop information. When the size of the viewport is small enough, the image is replaced with an image that is not cropped to size, therefore looking squashed and out of proportion.

**Comment by jeherve** [comment]
@chaitanyamsv Could you confirm that this is now fixed on WordPress.com, like in your example?

**Comment by georgestephanis** [comment]
@withinboredom The missing h and crop shouldn't be a problem, as Photon should just scale to the respective width and let the height match the original ratio.

**Comment by withinboredom** [comment]
Without the crop, here's some examples:

Without the h and crop:
![image](https://cloud.githubusercontent.com/assets/1883296/24375749/6f9d2c32-1307-11e7-9625-928b3c837fd7.png)

With the h and crop:
![image](https://cloud.githubusercontent.com/assets/1883296/24375765/7c5b0d54-1307-11e7-833e-9bc8722acf0b.png)

If you look carefully, `data-medium-file` and `data-large-file` aren't going through Photon as well.

**Comment by adennisphoto** [comment]
Hello all, I have no coding experience, just using the Sketch theme from Wordpress so relying on the theme to be corrected to sort this pixelated issue... Has anyone had any new revelations or can suggest another sophisticated way of showing photos in Sketch that does a similar thing without rendering my photos pixelated?

Many thanks

www.angela-dennis.com

**Comment by jeherve** [comment]
@adennisphoto We haven't made any progress on this just yet, but we'll update this issue when we do!

**Comment by cecoates** [comment]
I've got another report of the same thing:

https://michaelglancyglassworks.com/work/

This one also using the tiled gallery, but "square tiles".

321013-chat

**Comment by ryanmarkel** [comment]
I (apparently) ran into this problem today; this gallery is forced to using 700px-wide images in the carousel, and as a result, the carousel images look really bad even though there is a very high-res original available:

https://ryanmarkel.com/84139/destiny-2-month-one-gallery/

Access to the site is available upon request. :)

**Comment by liviopv** [comment]
Reported in #1336286-hc as well using the Sketch theme

**Comment by KZeni** [comment]
I'm not sure if we have exactly the same issue, but I was able to resolve this problem for myself with the following added to my functions.php file:
```
// Tell Jetpack that image galleries and the content width should be considered larger (make it so images within image galleries don't get shown too small [what tiled_gallery_content_width controls] as well as making sure the sizes="(max-width:***)" doesn't pull a size smaller than desired [what jetpack_content_width controls])
function jetpack_overwrite_image_width() {
	return 1920;
}
add_filter( 'tiled_gallery_content_width', 'jetpack_overwrite_image_width' );
add_filter( 'jetpack_content_width', ''jetpack_overwrite_image_width' );
```
This removed the sizes code from the <img> tag and other aspects within Jetpack's internals which had Photon serving up the incorrect size.

**Comment by nellofn** [comment]
A user reported this issue on their site using the Soho theme. It seems that the issue is present on these two blog posts:

(Purple flower)
https://esthergeerlingsphotography.com/2017/06/27/flowers/

(Imagine picture)
https://esthergeerlingsphotography.com/2016/08/02/nyc/

I'm curious that on both posts the affected image reported by the user is the first one, but it could be any image, I just can't tell whether it looks blurry or not.

Reported in: #1488255-hc
Follow up ticket: #890817-zen

**Comment by vparkhere** [comment]
Reported here:
https://csavargas.com/2018/02/16/porto-g/

Theme: Twenty Fifteen
#963651-zen

**Comment by joeboydston** [comment]
Another report in #1293358-zen
Theme: hemingway-rewritten

**Comment by donalirl** [comment]
Also reported in #446376-hc on the theme Blask

**Comment by jruud** [comment]
Also reported in #1639059-zen 
Theme: Canape

**Comment by coder-karen** [comment]
Potentially related: #1723608-zen (Jetpack site) using Blaskan theme

**Comment by StefMattana** [comment]
I believe Karen's ticket at #1723608-zen is indeed affected by this bug: they have the same problem even if switching to a default theme :(
I suggested the user a workaround to resize and reupload the images, and let's see how it goes.

**Comment by erricgunawan** [comment]
User in 1924944-zen is back and saying that they don't have the blurry issue anymore.

Although they're not really sure if they had anything to do with it.

They just add one new image to the media library and to the gallery, but didn't adjust anything in the existing galleries.

**Comment by dericleeyy** [comment]
1954009-zen

Any updates on when the fix for this will be implemented?

**Comment by jeherve** [comment]
@dericleeyy No updates, no. This issue will be updated once someone starts working on it. For now, we don't have an ETA for a fix.

**Comment by cecilearkay** [comment]
I have the opposite issue with my user. The images are crisp when opened in the carousel but in the gallery they are blurry. Would it be related? 

This is the gallery pic: https://clairecime.files.wordpress.com/2019/05/aperccca7u-expo-aufi.png?w=1478 (blurry)
This is the carousel pic: https://clairecime.files.wordpress.com/2019/05/aperccca7u-expo-aufi.png?w=1153&h=&zoom=2 (crisp)

**Comment by jeherve** [comment]
@cecilearkay This is most likely not related, no. WordPress.com uses its own custom solution to handle image resizing, and does not rely on Photon (used on Jetpack sites).

I would recommend opening a ticket on wpcom.trac for your issue.

**Comment by damiannep** [comment]
User reported this issue also in #5822729-hc. They would like to know when the issue is resolved.

**Comment by chaitanyamsv** [comment]
Another report in #12938692-hc

**Comment by pjtfernandes** [comment]
Hi, I am also experiencing this issue on my page.
https://www.blushingbride.co.za/weddingdresses/a-line/caitlin/

**Comment by KZeni** [comment]
Is this a matter of a bug/shortcoming in the themes being used as they should use what I proposed at https://github.com/Automattic/jetpack/issues/6625#issuecomment-354378034 to accommodate Jetpack's Photon functionality rather than being an issue with Jetpack itself? Or is this actually a but that Jetpack itself should look to avoid? I'm thinking it's the former since it's currently intentionally grabbing `$content_width` for this max image size and has hooks to overwrite this behavior as needed (per my comment above & what I linked to here).

It seems like Jetpack has made a decision on how this acts & it's acting as expected. This thread is effectively people reporting problems with the themes they're using not playing nicely with Jetpack (unless something regarding how Jetpack intends to handle this ends up changing.)
